arangod/VocBase/LogicalCollection.cpp

```diff
@@ -1282,20 +1278,19 @@ void LogicalCollection::decorateWithInternalValidators() {
 }
 
 replication2::LogId LogicalCollection::shardIdToStateId(
-    std::string_view shardId) {
+    ShardID const& shardId) {
   auto logId = tryShardIdToStateId(shardId);
   ADB_PROD_ASSERT(logId.has_value())
       << " converting " << shardId << " to LogId failed";
   return logId.value();
 }
 
 std::optional<replication2::LogId> LogicalCollection::tryShardIdToStateId(
-    std::string_view shardId) {
-  if (shardId.empty()) {
+    ShardID const& shardId) {
+  if (!shardId.isValid()) {
     return {};
   }
-  auto stateId = shardId.substr(1, shardId.size() - 1);
-  return replication2::LogId::fromString(stateId);
+  return replication2::LogId::fromString(std::to_string(shardId.id()));
```

Comment:
This looks a bit strange. We are generating a temporary string from the shard number (without the "s"), and then are parsing that temporary string again.
Can't we use `return replication2::LogId(shardId.id());` here?